packages/react-native-reanimated/src/layoutReanimation/defaultTransitions/FadingTransition.ts

```diff
@@ -29,7 +29,8 @@ export class FadingTransition
     const delayFunction = this.getDelayFunction();
     const callback = this.callbackV;
     const delay = this.getDelay();
-    const duration = (this.durationV ?? 500) / 2;
+    const duration = this.durationV ?? 500;
+    const halfDuration = duration / 2;
```

Comment:
```suggestion
    const halfDuration = (this.durationV ?? 500) / 2;
```

We don't want to use `duration` anywhere because `originX`, `originY`, `width`, and `height` should be changed in the middle of the animation when the component becomes invisible. Therefore, the animation should look like this:

```js
originX: withDelay(
  delay + halfDuration,
  withTiming(values.targetOriginX, { duration: 0 })
),
```